flocker/node/_config.py

```diff
@@ -1245,35 +1248,50 @@ def deployment_from_configuration(deployment_configuration, all_applications):
     return set(nodes)
 
 
-def model_from_configuration(applications, deployment_configuration):
+def parse_desired_configuration(application_configuration,
+                                deployment_configuration):
     """
     Validate and coerce the supplied application configuration and
     deployment configuration dictionaries into a ``Deployment`` instance.
 
-    :param dict applications: Map of application names to ``Application``
-        instances.
+    These dictionaries are parsed from the YAML files passed to
+    ``flocker-deploy``.
 
-    :param dict deployment_configuration: Map of node names to application
-        names.
+    :param dict applications: Marshalled application configuration as
+         passed to ``flocker-deploy``.
+
+    :param dict deployment_configuration: Marshalled deployment
+         configuration as passed to ``flocker-deploy``.
 
     :raises ConfigurationError: if there are validation errors.
 
     :returns: A ``Deployment`` object.
     """
-    nodes = deployment_from_configuration(
+    fig_configuration = FigConfiguration(application_configuration)
+    if fig_configuration.is_valid_format():
+        applications = fig_configuration.applications()
+    else:
+        configuration = FlockerConfiguration(application_configuration)
+        if configuration.is_valid_format():
+            applications = configuration.applications()
+        else:
+            raise ConfigurationError(
+                "Configuration is not a valid Fig or Flocker format.")
+
+    nodes = _apply_deployment_to_applications(
         deployment_configuration, applications)
     return Deployment(nodes=frozenset(nodes))
 
 
-def current_from_configuration(current_configuration):
+def parse_current_cluster_state(current_cluster_state):
```

Comment:
As long as we stick with `Deployment`, mixing the word "cluster" in here is probably confusing.  So `parse_current_deployment_state` at a minimum - but even better, I think, `Deployment.from_state`?  Or maybe `Deployment.from_current_state` (but while "current" seems to clarify things, I think it actually obscures the fact that there's really no way to know when this state was present on the cluster and we just **hope** it's still the same as the current state).